### PR TITLE
USWDS-SITE-962 Tooltip Guidance

### PR DESCRIPTION
**Pull request by jaredcunha**
This pull request creates a guidance page for using tooltips. 

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `npm test` and make sure the tests for the files you have changed have passed.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Comment by jaredcunha** [comment]
Screen Shot 2020-05-27 at 11 30 56 AM

Is this the example we want to show? I was wondering if we should show eah button on its own row, even if that means adding more vertical weight before getting into the guidance. 

Was also unsure on what might be the best approach to showing multiple utilities applied in the last example.

**Comment by jaredcunha** [comment]
Component PR: https://github.com/uswds/uswds/pull/3485

File: _components/tooltips.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <li>By default, tooltips will appear on top of the element that triggers them</li>
+        <li>Use the <code>data-position</code> attribute to indicate where tooltip appears in relation to the element that makes it appear:
+          <ul>
+            <li><code>data-position="top"</code>: On top, horizontally centered. If the <code>data-position</code> attribute is omitted, the tooltip will appear on top by default.</li>
+            <li><code>data-position="bottom"</code>: Below, horizontally centered</li>
+            <li><code>data-position="right"</code>: To the right, vertically centered</li>
+            <li><code>data-position="left"</code>: To the left, vertically centered</li>
+          </ul>
+        </li>
+        <li>Tooltips are protected from clipped by the viewport. If clipping is detected, the tooltip is positioned to the opposite side indicated in the <code>data-position</code> attribute. If the tooltip is still clipped, it placed on top of the element, with its width constrained to the width of the triggering element. Only then does the tooltip wrap to multiple lines.</li>
```
Comment (thisisdano) [inline _components/tooltips.md:60]: - `from clipped by the viewport` → `viewport clipping`
- `to the opposite side indicated in the` → `on the opposite side`
- `it placed on` → `it is positioned on`

File: _components/tooltips.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        <li>Most of the tooltip’s elements are generated on the page by JavaScript. In order to apply <a href="{{ site.baseurl }}/utilities/">utility classes</a> to the tooltip’s wrapping element, place them inside a <code>data-classes</code> attribute, separated by spaces.</li>
```
Comment (thisisdano) [inline _components/tooltips.md:61]: remove "on the page"

File: _components/tooltips.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+---
+permalink: /components/tooltips/
+layout: styleguide
+title: Tooltips
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Switched to singular

File: _components/tooltips.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+---
+permalink: /components/tooltips/
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Switched to singular

File: _components/tooltips.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        <li><strong>Brief descriptions.</strong> Tooltips should be concise, only a few words.</li>
+        <li><strong>Helpful, but not critical information.</strong></li>
+        <li><strong>Enhance confidence.</strong> Tooltips can be used to increase certainty about an interaction.</li>
+        <li><strong>Lack of space.</strong> Tooltips can be helpful when your UI can be short on space. All other options should be exhausted for keeping content visible on the page.</li>
+      </ul>
+      <h4>When to consider something else</h4>
+      <ul class="usa-content-list">
+        <li><strong>Lengthy descriptions.</strong> Tooltips are micro-content, they should be brief. If you need a lot of text, consider an alternative approach.</li>
+        <li><strong>Vital information.</strong> Information that is important for completing a task should never be hidden behind an interaction.</li>
+        <li><strong>Redundant content.</strong> Do not use where the information is repetitive or usability is obvious.</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: "is is repetitive" typo made LOL!

File: _components/tooltips.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      <ul class="usa-content-list">
+        <li><strong>Brief descriptions.</strong> Tooltips should be concise, only a few words.</li>
+        <li><strong>Helpful, but not critical information.</strong></li>
+        <li><strong>Enhance confidence.</strong> Tooltips can be used to increase certainty about an interaction.</li>
+        <li><strong>Lack of space.</strong> Tooltips can be helpful when your UI can be short on space. All other options should be exhausted for keeping content visible on the page.</li>
+      </ul>
+      <h4>When to consider something else</h4>
+      <ul class="usa-content-list">
+        <li><strong>Lengthy descriptions.</strong> Tooltips are micro-content, they should be brief. If you need a lot of text, consider an alternative approach.</li>
+        <li><strong>Vital information.</strong> Information that is important for completing a task should never be hidden behind an interaction.</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      <h4>When to use the tooltip component</h4>
+      <ul class="usa-content-list">
+        <li><strong>Brief descriptions.</strong> Tooltips should be concise, only a few words.</li>
+        <li><strong>Helpful, but not critical information.</strong></li>
+        <li><strong>Enhance confidence.</strong> Tooltips can be used to increase certainty about an interaction.</li>
+        <li><strong>Lack of space.</strong> Tooltips can be helpful when your UI can be short on space. All other options should be exhausted for keeping content visible on the page.</li>
+      </ul>
+      <h4>When to consider something else</h4>
+      <ul class="usa-content-list">
+        <li><strong>Lengthy descriptions.</strong> Tooltips are micro-content, they should be brief. If you need a lot of text, consider an alternative approach.</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  <div class="usa-accordion usa-accordion--bordered site-accordion-docs">
+    <button class="usa-button-unstyled usa-accordion__button"
+        aria-expanded="true" aria-controls="tooltip-docs">
+      Guidance
+    </button>
+    <div id="tooltip-docs" aria-hidden="false" class="usa-accordion__content site-component-usage">
+      <h4>When to use the tooltip component</h4>
+      <ul class="usa-content-list">
+        <li><strong>Brief descriptions.</strong> Tooltips should be concise, only a few words.</li>
+        <li><strong>Helpful, but not critical information.</strong></li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  {% include code/accordion.html component="tooltips" %}
+  <div class="usa-accordion usa-accordion--bordered site-accordion-docs">
+    <button class="usa-button-unstyled usa-accordion__button"
+        aria-expanded="true" aria-controls="tooltip-docs">
+      Guidance
+    </button>
+    <div id="tooltip-docs" aria-hidden="false" class="usa-accordion__content site-component-usage">
+      <h4>When to use the tooltip component</h4>
+      <ul class="usa-content-list">
+        <li><strong>Brief descriptions.</strong> Tooltips should be concise, only a few words.</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        <li><strong>Helpful, but not critical information.</strong></li>
+        <li><strong>Enhance confidence.</strong> Tooltips can be used to increase certainty about an interaction.</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+      </ul>
+      <h4 class="usa-heading">Accessibility</h4>
+      <ul class="usa-content-list">
+        <li><strong>Use as title attribute.</strong> Tooltips are essentially a progressive enhancement for a title attribute, and will gracefully degrade to it if the component does not intialize.</li>
+        <li><strong>Keyboard accessibility</strong> Tooltips make title attributes keyboard accessible.</li>
+      </ul>
+      <h4 class="usa-heading">Implementation</h4>
+      <ul class="usa-content-list">
+        <li>Any element given the class name <code>usa-tooltip</code> and a <code>title</code> attribute will become a tooltip.</li>
+        <li>Place tooltips on elements that contain little or no child elements</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        <li>Elements or text that show a tooltip when hovered or focused will be prevented wrapping onto a new line and will be given <code>tabindex="0"</code> for keyboard interaction.</li>
+        <li>By default, tooltips will appear on top of the element that triggers them</li>
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        <li>Use the <code>data-position</code> attribute to indicate where tooltip appears in relation to the element that makes it appear:
```
[the remark this answers is not in the document]
Comment (jaredcunha) [reply _components/tooltips.md]: Done

File: _components/tooltips.md (hunk 1 above)
Comment (jaredcunha) [reply _components/tooltips.md]: Done.

File: _components/tooltips.md (hunk 2 above)
Comment (jaredcunha) [reply _components/tooltips.md]: Done

**Comment by jaredcunha** [comment]
@thisisdano i've worked through your suggestions.

File: _components/tooltip.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+        aria-expanded="true" aria-controls="tooltip-docs">
+      Guidance
+    </button>
+    <div id="tooltip-docs" aria-hidden="false" class="usa-accordion__content site-component-usage">
+      <h4>When to use the tooltip component</h4>
+      <ul class="usa-content-list">
+        <li><strong>Helpful, but not critical information.</strong></li>
+        <li><strong>Enhance confidence.</strong> Tooltips can be used to increase certainty about an interaction.</li>
+        <li><strong>Brief descriptions.</strong> Tooltips can be a good choice when the the helper text is only a few words long.</li>
+        <li><strong>Lack of space.</strong> Tooltips can be helpful when your UI can be short on space. All other options should be exhausted for keeping content visible on the page.</li>
```
Comment (Ben-Judy-Flexion) [inline _components/tooltip.md:25]: Suggested rewrite: "Tooltips can be helpful when your UI is short on space."

**Comment by jaredcunha** [comment]
Replaced with https://github.com/uswds/uswds-site/pull/963

**Pull request closed without merging**